docs/how-tos/nexus-validation-dev-notes.md

```diff
@@ -0,0 +1,473 @@
+# NeXus Validation Dev. Notes
```

Comment:
This could eventually go to the validation docs page if you split them as suggested above.

Reply:
I feel like the information in this file is pretty useful, but it needs to be heavily cleaned before going to the official docs.

Reply:
yea. Right, I was not sure what do to with it. It is important for us, to see what problems are related to these validation methods. I also have to add an part about almost empty NeXus files, as only nxvalidate points out missing required nexus concepts. 
I intended this to be for us internal, to point out current problems, which most probably have to be solved in the long run by verify_nexus, as the other projects are barely maintained.